### ISSUE TITLE
Scroll to a pageYOffset value

### ISSUE THREAD

**Issue by Aweary** [cferdinandi/smooth-scroll#210]
Hey there,

I couldn't find this anywhere in the docs: is it possible to scroll to a `pageYOffset` value, instead of passing in CSS selectors/elements?

I'd just really to do something like `smoothScroll.animateTo(1400)`  or something similar. Is that supported?

It seems like it'd be as easy as allowing a `scrollPosition` property in the `options` argument of `animateScroll` and using that value as the `endLocation` if it exists. Or you could do a type check on the first argument, `toggle`.

```js
smoothScroll.animateScroll = function ( toggle, anchor, options ) {
  if (typeof toggle === "number") {
     /* Just scroll to that point */
  }
```

 If this feature doesn't exist yet, and you'd be OK with it existing, I'd be happy to get a PR going.

**Comment by cferdinandi** [comment]
Hi there! So that's not supported, BUT... it's worth noting that you can pass in `toggle` as null and simply specify the ID of the element you want to scroll to for `anchor`:

```js
smoothScroll.animateScroll( null, '#some-element' );
```

Not exactly the same as passing in just a number, but I think ultimately achieves what most people would be hoping to accomplish with this script. I'm open to considering adding the numeric option, but I'd like to better understand the use case. Could you expand on it a bit?

**Comment by Aweary** [comment]
My use case involves a React application where the position of an element may change depending on the `pageYOffset` value, and I'd like to just pass the computed value to `animateScroll` instead of having to apply it to the element, query the element again, and call `animateScroll`.

Also, since React is managing my DOM for me I'd like to avoid having to add any IDs manually. That would involve using `refs` and `ReactDOM` to query the element directly which is less than ideal.

Basically instead of having to do `computeNewPosition => applyToDOM => queryDOM => scrollToDOM`
I'd like to just `computeNewPosition => applytoDOM => scrollToNewPosition`

This would make `smooth-scroll` more flexible since it could allow the computation of the scroll position to be offloaded to another framework/library (React in this case) as appropriate.

**Comment by cferdinandi** [comment]
Solid. Makes perfect sense! I'm totally down with this as a PR, but let's just make sure that `anchor` is the where you pass in the number (instead of `toggle`).

**Comment by Aweary** [comment]
Sounds good, I'll get a PR going :+1:

**Comment by cferdinandi** [comment]
Quick thought (and I can make this update if you'd prefer): since you'll always need a thing to scroll to (the `anchor`), but won't always pass in a `toggle`, should the order of those arguments get reversed so that you can avoid passing in null when just using a number of ID?

Example:

```js
smoothScroll.animateScroll = function ( anchor, toggle, options ) { ... }
```

**Comment by Aweary** [comment]
I think that makes sense, arguments should be ordered by importance, so if `toggle` would be conditional and `anchor` always required, `anchor` should come first :+1: I think I can tackle that as well, it doesn't look like its a huge effort.

**Comment by cferdinandi** [comment]
No obligation. If it's a pain, just leave it and I'll knock it out.

**Comment by cferdinandi** [comment]
Hi there. I wanted to rewrite the script a bit to add some other features, so I tossed support for this in while I was at it. All set: https://github.com/cferdinandi/smooth-scroll/pull/215

**Comment by Aweary** [comment]
Great! Thanks so much.

**Comment by cferdinandi** [comment]
Also, check the updated docs, since the `animateScroll` variables reversed order a bit...
